swarmrl/utils/utils.py

```diff
@@ -271,3 +272,36 @@ def record_training(training_dict: dict):
         training_data,
         allow_pickle=True,
     )
+
+
+def calc_signed_angle_between_directors(my_director, other_director):
+    """
+    In 2D compare two different normalized
+    directors to determine the angle between them
+
+    Parameters
+    ----------
+
+    my_director : np.ndarray
+            Normalised director in 3D.
+    other_director : np.ndarray
+            Normalised director in 3D.
+    Returns
+    singed float which represents the signed angle of my_director to otherdirector
+    with the mathematical sign convention.
+    """
+
+    # Assert if the directors were really normalized
+    # np.testing.assert_almost_equal(jnp.linalg.norm(my_director), 1, decimal=6)
```

Comment:
The assertions are a nice idea, but instead of raising an error, maybe try the following:

```python
if np.linalg.norm(my_director) != 1:
    my_director /= np.linalg.norm(my_director)
```
In the case where you cannot have an if statement here, e.g. traced jax code, just perform the normalization as the overhead such as:

```python
my_director /= np.linalg.norm(my_director)
```

Reply:
I disagree. we should just check that the system is 2d because the observable only works in 2d. directors never need to be normalized because they are by default

Reply:
If the director is normalised by default then sure, there is no need for the assertion or a default normalisation procedure and we can avoid the problem. Happy to remove that.